Source/CoreExtensions/Private/CoreExtWaitForPrimaryPawn.cpp

```diff
@@ -0,0 +1,89 @@
+#include "CoreExtWaitForPrimaryPawn.h"
+
+#include "GameFramework/GameModeBase.h"
+
+#include <Engine/GameInstance.h>
+
+UCoreExtWaitForPrimaryPawn * UCoreExtWaitForPrimaryPawn::WaitForPrimaryPawn( UObject * world_context_object )
+{
+    ThisClass * action = nullptr;
+
+    if ( auto * world = GEngine->GetWorldFromContextObject( world_context_object, EGetWorldErrorMode::LogAndReturnNull ) )
+    {
+        action = NewObject< ThisClass >();
+        action->WorldPtr = world;
+        action->RegisterWithGameInstance( world );
+    }
+    return action;
+}
+
+void UCoreExtWaitForPrimaryPawn::Activate()
+{
+    Super::Activate();
+
+    if ( const auto * game_instance = WorldPtr.Get()->GetGameInstance() )
+    {
+        if ( auto * player_controller = game_instance->GetPrimaryPlayerController() )
```

Comment:
Cache player controller in member variable
You can then reuse it in the SetReadyToDestroy

Reply:
Done